cranelift/isle/isle/src/codegen.rs

```diff
@@ -499,167 +508,195 @@ impl<L: Length, C> Length for ContextIterWrapper<L, C> {{
+                        }
 
-                &ControlFlow::Return { pos, result } => {
-                    writeln!(
-                        ctx.out,
-                        "{}// Rule at {}.",
-                        &ctx.indent,
-                        pos.pretty_print_line(&self.typeenv.filenames)
-                    )?;
-                    write!(ctx.out, "{}", &ctx.indent)?;
-                    match ret_kind {
-                        ReturnKind::Plain | ReturnKind::Option => write!(ctx.out, "return ")?,
-                        ReturnKind::Iterator => write!(ctx.out, "returns.extend(Some(")?,
-                    }
-                    self.emit_expr(ctx, result)?;
-                    if ctx.is_ref.contains(&result) {
-                        write!(ctx.out, ".clone()")?;
-                    }
-                    match ret_kind {
-                        ReturnKind::Plain | ReturnKind::Option => writeln!(ctx.out, ";")?,
-                        ReturnKind::Iterator => {
-                            writeln!(ctx.out, "));")?;
+                        &ControlFlow::Return { pos, result } => {
                             writeln!(
                                 ctx.out,
-                                "{}if returns.len() >= MAX_ISLE_RETURNS {{ return; }}",
-                                ctx.indent
+                                "{}// Rule at {}.",
+                                &ctx.indent,
+                                pos.pretty_print_line(&self.typeenv.filenames)
                             )?;
+                            write!(ctx.out, "{}", &ctx.indent)?;
+                            match ret_kind {
+                                ReturnKind::Plain | ReturnKind::Option => {
+                                    write!(ctx.out, "return ")?
+                                }
+                                ReturnKind::Iterator => write!(ctx.out, "returns.extend(Some(")?,
+                            }
+                            self.emit_expr(ctx, result)?;
+                            if ctx.is_ref.contains(&result) {
+                                write!(ctx.out, ".clone()")?;
+                            }
+                            match ret_kind {
+                                ReturnKind::Plain | ReturnKind::Option => writeln!(ctx.out, ";")?,
+                                ReturnKind::Iterator => {
+                                    writeln!(ctx.out, "));")?;
+                                    writeln!(
+                                        ctx.out,
+                                        "{}if returns.len() >= MAX_ISLE_RETURNS {{ return; }}",
+                                        ctx.indent
+                                    )?;
+                                }
+                            }
                         }
                     }
                 }
+
+                Nested::Arms(source, arms) => {
```

Comment:
I can't seem to find anything that pops the `Nested::Arms` entries from the stack -- does that ever happen? If not, is that a bug? I would assume it is?